app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -153,6 +158,19 @@ fun ActivityDetailsScreen(
     listActivityViewModel.updateActivity(activity!!.copy(comments = comments))
   }
 
+  val creatorID = activity?.creator ?: ""
+  var creator by remember {
+    mutableStateOf(User(creatorID, "anonymous", "anonymous", listOf(), listOf(), "", listOf()))
+  }
+  activity?.participants?.forEach() {
+    if (it.id == creatorID) {
+      creator = it
+    }
+  }
```

Comment:
You continue running the list even if you found the creator, my suggestion shows a way of stopping once a creator is found. For small activities, your implementation should not be harmful, but if we imagine that there are hundreds of participants and the creator is the first participant, it willl run all the list for nothing
```suggestion
activity?.participants?.firstOrNull { it.id == creatorID }?.let {
    creator = it
}
```